src/wallet/wallet2.cpp

```diff
@@ -3465,6 +3466,18 @@ void wallet2::refresh(bool trusted_daemon, uint64_t start_height, uint64_t & blo
           throw std::runtime_error("proxy exception in refresh thread");
       }
 
+      if (!next_blocks.empty() && !m_blockchain.is_in_bounds(next_blocks_start_height))
+      {
+        const uint64_t expected_start_height = std::max(blocks_start_height, uint64_t(1)) - 1;
+        const uint64_t reorg_depth = expected_start_height - std::min(expected_start_height, next_blocks_start_height);
+        if (reorg_depth > m_max_reorg_depth)
+        {
+          MERROR("Rejected " << reorg_depth << " blocks reorg (max allowed " << m_max_reorg_depth << ")"
+            << ", set max-reorg-depth " << reorg_depth << " if you want to allow it.");
+          throw std::runtime_error("reorg depth exceeds maximum allowed");
```

Comment:
The option suggestion should probably be in the exception, walet users won't see logs.

Reply:
Updated, introduced new exception class for that